packages/financial-templates-lib/src/helpers/GasEstimator.ts

```diff
@@ -255,9 +255,8 @@ export class GasEstimator {
       if (etherscanGasResponse.result.suggestBaseFee === undefined)
         throw new Error(`Bad ethgasstation response ${json}`);
       return {
-        maxFeePerGas: Number(etherscanGasResponse.result.FastGasPrice),
-        maxPriorityFeePerGas:
-          Number(etherscanGasResponse.result.FastGasPrice) - Number(etherscanGasResponse.result.suggestBaseFee),
+        maxFeePerGas: Number(etherscanGasResponse.result.suggestBaseFee) * 2,
+        maxPriorityFeePerGas: Number(etherscanGasResponse.result.suggestBaseFee),
```

Comment:
This will be a very high priority fee, at least for mainnet (not as sure about polygon). What's the rationale for setting the priority fee to the suggested base fee?

Reply:
@mrice32 I think we decided to do maxFeePerGas = 2 * base fee, actually I've checked and it was 3 times.

Given that `maxFeePerGas = baseFeePerGas + maxPriorityFeePerGas` it means that `maxPriorityFeePerGas` should be `baseFeePerGas`

Do you agree?

Reply:
I'm not sure I agree because of the maxes.

feePerGas = baseFeePerGas + priorityFeePerGas

The way I think about the maxes is that they put a cap on each of the values. We are allowed to cap the total fee and the priority fee. The difference between those caps is how high the base fee can get (but it can always be lower, meaning you'll pay less!).

Setting a high maxFeePerGas, but a fairly low maxPriorityFeePerGas just means that you are constraining the tip (tips rarely need to be higher than 1 gwei or so), but allowing the algorithmic base bee (burned amount) to fluctuate without blocking your transaction.

I would suggest either setting a hardcoded maxPriorityFeePerGas or using an API value that you are confident will not make it very high. High tips (maxPriorityFeePerGas) are basically just giving away money, but high maxFeePerGas (room for the base fee to fluctuate) is what stops your txn from getting blocked bc it is impossible to include.

TLDR: maxPriorityFeePerGas should always be less than 2 gwei (even values as low as 0.1 gwei are probably fine!). maxFeePerGas should be 2-3x current base fee.

Does that make sense?

Reply:
Thanks @mrice32,
I've updated the calculations to set the `maxFeePerGas` to be 3 x the baseFee and hardcoded the `maxPriorityFeePerGas` in mainnet and polygon (5 gwei and 50gwei). I've decided the hardcoded values by looking at priority values showed here: https://etherscan.io/gastracker (3~6 gwei) and here https://polygonscan.com/gastracker (30~70 gwei)